tests/test_api.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/python3 -OO
+# Copyright 2007-2020 The SABnzbd-Team <team@sabnzbd.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+"""
+tests.test_api - Tests for API functions
+"""
+
+from tests.testhelper import *
+
+import sabnzbd.api as api
+import sabnzbd.version
+
+
+class TestApiInternals:
+    """ Test internal functions of the API """
+
+    def test_empty(self):
+        with pytest.raises(TypeError):
+            api.api_handler(None)
+        with pytest.raises(AttributeError):
+            api.api_handler("")
+
+    @set_config({"disable_key": False})
+    def test_mode_invalid(self):
+        expected_error = "error: API Key Required"
+        assert api.api_handler({"mode": "invalid"}).strip() == expected_error
+        with pytest.raises(IndexError):
+            assert api.api_handler({"mode": []}).strip() == expected_error
+            assert api.api_handler({"mode": ""}).strip() == expected_error
+            assert api.api_handler({"mode": None}).strip() == expected_error
+
+    def test_version(self):
+        assert api.api_handler({"mode": "version"}).strip() == sabnzbd.__version__
+
+    @set_config({"disable_key": False})
+    def test_auth(self):
+        assert api.api_handler({"mode": "auth"}).strip() == "apikey"
```

Comment:
Should we just add `.strip` to the `api_handler`?

Reply:
The api code explicitly adds those newlines, so they may well serve some purpose. Maybe simply that the receiving end can be sure they read the entire line, rather than partial output?

Reply:
Aaah, ok!